ui/apps/dashboard/src/components/Metrics/Dashboard.tsx

```diff
@@ -0,0 +1,143 @@
+import { useQuery } from 'urql';
+
+import { GetBillingPlanDocument } from '@/gql/graphql';
+import { FailedFunctions } from './FailedFunctions';
+import { FunctionStatus } from './FunctionStatus';
+
+type EntityType = {
+  id: string;
+  name: string;
+};
+
+export const DEFAULT_DURATION = { hours: 24 };
+
+const getFrom = (start?: Date, duration?: DurationType | '') =>
+  start || subtractDuration(new Date(), duration ? duration : DEFAULT_DURATION);
+
+const getDefaultRange = (start?: Date, end?: Date, duration?: DurationType | '') =>
+  start && end
+    ? {
+        type: 'absolute' as const,
+        start: start,
+        end: end,
+      }
+    : {
+        type: 'relative' as const,
+        duration: duration ? duration : DEFAULT_DURATION,
+      };
+
+export const Dashboard = ({
+  apps = [],
+  functions = [],
+}: {
+  apps?: EntityType[];
+  functions?: EntityType[];
+}) => {
+  const [selectedApps, setApps, removeApps] = useStringArraySearchParam('apps');
+  const [selectedFns, setFns, removeFns] = useStringArraySearchParam('fns');
+  const [start] = useSearchParam('start');
+  const [end] = useSearchParam('end');
+  const [duration] = useSearchParam('duration');
+  const [autoRefresh] = useBooleanSearchParam('autoRefresh');
+  const batchUpdate = useBatchedSearchParams();
+
+  const parsedDuration = duration && parseDuration(duration);
+  const parsedStart = toDate(start);
+  const parsedEnd = toDate(end);
+
+  const [overviewOpen, setOverviewOpen] = useState(true);
+
+  const [{ data: planData }] = useQuery({
+    query: GetBillingPlanDocument,
+  });
+
+  const logRetention = Number(planData?.account.plan?.features.log_retention);
+  const upgradeCutoff = subtractDuration(new Date(), { days: logRetention || 7 });
+
+  return (
+    <div className="flex h-full w-full flex-col">
+      <div className="bg-canvasBase flex h-16 w-full flex-row items-center justify-between px-3 py-5">
+        <div className="flex flex-row items-center justify-start gap-x-2">
+          <EntityFilter
+            type="app"
+            onFilterChange={(apps) => (apps.length ? setApps(apps) : removeApps())}
+            selectedEntities={selectedApps || []}
+            entities={apps}
+            className="h-8"
+          />
+          <EntityFilter
+            type="function"
+            onFilterChange={(fns) => (fns.length ? setFns(fns) : removeFns())}
+            selectedEntities={selectedFns || []}
+            entities={functions}
+            className="h-8"
+          />
+        </div>
+        <div className="flex flex-row items-center justify-end gap-x-2">
+          <Pill appearance="outlined" kind="warning">
+            <div className="text-nowrap">15m delay</div>
+          </Pill>
+          <RangePicker
+            className="w-full"
+            upgradeCutoff={upgradeCutoff}
+            defaultValue={getDefaultRange(parsedStart, parsedEnd, parsedDuration)}
+            onChange={(range: RangeChangeProps) => {
```

Comment:
I serialize absolute range and relative offset differently because ux has a refresh built in which implies a rolling window when the user has selected a relative offset.